File: binary_search.py


def binarySearch(arr, l, u, x):
    
    if (u >= l):
        mid = l + (u - l) // 2
        if(arr[mid] == x):
            return mid
            
        elif (arr[mid] > x):
            return binarySearch(arr, l, mid-1, x)

        else:
            return binarySearch(arr, mid+1, u, x)

    else:
        return -1

arr = [1,2,3,4,5]
x = 3
l = 0
result = binarySearch(arr, l, len(arr)-1, x)
if (result != -1):
    print("The index of the searched number is at: %d" % result)
else:
    print("Searched number not found")

File: binary_search.c
#include<stdio.h>


int binarySearch(int arr[], int lo, int hi, int x){
    if (hi>=lo){
        int mid = lo + (hi - lo) / 2;
        if (arr[mid] == x)
            return mid;


        if (arr[mid] > x)
            return binarySearch(arr, lo, mid-1, x);
        

        return binarySearch(arr, mid+1, hi, x);
        
    return -1;
    }
}


int main(void)
{
    int n, x, arr[50];

    printf("\nEnter the number of elements: ");
    scanf("%d",&n);
    for (int i=0; i<n; ++i){
        printf("\nEnter the %d element: ", i+1);
        scanf("%d",&arr[i]);
    }
    printf("\nEnter the element to search: ");
    scanf("%d",&x);

    int result = binarySearch(arr, 0 , n-1, x);
    (result == -1) ? printf("\nElement not present in the array\n")
                   : printf("\nElement present at index %d\n", result);

    return 0;
}
